site/src/xServices/createTemplate/createTemplateXService.ts

```diff
@@ -0,0 +1,393 @@
+            data: TemplateExample
+          }
+          createFirstVersion: {
+            data: TemplateVersion
+          }
+          createVersionWithParameters: {
+            data: TemplateVersion
+          }
+          waitForJobToBeCompleted: {
+            data: TemplateVersion
+          }
+          loadParameterSchema: {
+            data: ParameterSchema[]
+          }
+          createTemplate: {
+            data: Template
+          }
+        },
+      },
+      tsTypes: {} as import("./createTemplateXService.typegen").Typegen0,
+      initial: "starting",
+      states: {
+        starting: {
+          always: [
+            { target: "loadingStarterTemplate", cond: "isExampleProvided" },
+            { target: "idle" },
+          ],
+          tags: ["loading"],
+        },
+        loadingStarterTemplate: {
+          invoke: {
+            src: "loadStarterTemplate",
+            onDone: {
+              target: "idle",
+              actions: ["assignStarterTemplate"],
+            },
+            onError: {
+              target: "idle",
+              actions: ["assignError"],
+            },
+          },
+          tags: ["loading"],
+        },
+        idle: {
+          on: {
+            CREATE: {
+              target: "creating",
+              actions: ["assignTemplateData"],
+            },
+            UPLOAD_FILE: {
+              actions: ["assignFile"],
+              target: "uploading",
+              cond: "isNotUsingExample",
+            },
+            REMOVE_FILE: {
+              actions: ["removeFile"],
+              cond: "hasFile",
+            },
+          },
+        },
+        uploading: {
+          invoke: {
+            src: "uploadFile",
+            onDone: {
+              target: "idle",
+              actions: ["assignUploadResponse"],
+            },
+            onError: {
+              target: "idle",
+              actions: ["displayUploadError", "removeFile"],
+            },
+          },
+        },
+        creating: {
+          initial: "creatingFirstVersion",
+          states: {
+            creatingFirstVersion: {
+              invoke: {
+                src: "createFirstVersion",
+                onDone: {
+                  target: "waitingForJobToBeCompleted",
+                  actions: ["assignVersion"],
+                },
+                onError: {
+                  actions: ["assignError"],
+                  target: "#createTemplate.idle",
+                },
+              },
+              tags: ["submitting"],
+            },
+            waitingForJobToBeCompleted: {
+              invoke: {
+                src: "waitForJobToBeCompleted",
+                onDone: [
+                  {
+                    target: "loadingMissingParameters",
+                    cond: "hasMissingParameters",
+                    actions: ["assignVersion"],
+                  },
+                  {
+                    target: "#createTemplate.idle",
```

Comment:
This `#createTemplate.idle` goes back up to line 115, right?

Reply:
Yes